packages/neon-core/src/sc/ContractParam.ts

```diff
@@ -16,16 +16,31 @@ export enum ContractParamType {
   Signature = 0x17, // TODO: Implement support
 
   Array = 0x20,
-  Map = 0x22, // TODO: Implement support
+  Map = 0x22,
 
   InteropInterface = 0x30, // TODO: Implement support
 
   Void = 0xff,
 }
+export type ContractParamMapLike = {
+  key: ContractParamLike;
+  value: ContractParamLike;
+}[];
+
+export type ContractParamMap = {
+  key: ContractParam;
+  value: ContractParam;
+}[];
 
 export interface ContractParamJson {
   type: string;
-  value?: string | boolean | number | ContractParamJson[] | null;
+  value?:
+    | string
+    | boolean
+    | number
+    | ContractParamJson[]
+    | ContractParamMapLike

```

Comment:
The json representation should only include json stuff. So you will need a `ContractParamMapJson` type here. (The usage of the *Json types is at the serialization boundary so we should be assured that the input is always plain json)